### PR TITLE
arm and m1 apple silicon support

### PR DESCRIPTION
**Pull request by vdovhanych**
This introduces support for M1 or any other arm device that supports docker. This pr is still a draft since there need to be some more changes. 

TODO: 

- [x] Add arm device in our CI for building the emulator 
- [x] Add arm device in our CI for building arm Docker image. Alternatively, set it up with `docker buildx`
- [x] Change firmware script to fetch latest arm emulator build from CI

### CODE REVIEW

**Comment by tsusanka** [comment]
OMG IT WORKS 🎉 🎉 🎉 

image

**Comment by tsusanka** [comment]
I have dared to modify the docs a bit :).

**Comment by vdovhanych** [comment]
Let's close this 😢. It should work out of the box now with the latest docker app for m1, so no need for this 🙈

**Comment by tsusanka** [comment]
Upon further discussion we came to the conclusion this will be more stable and makes sense to finish. We will not worry about old firmwares: we will start supporting ARM from the latest release and if one needs older FWs they simply run it via rosette.

**Comment by vdovhanych** [comment]
This should be finished and ready to be merged with a tiny exception. We need to change the download link for master emulators once the changes are merged to master there. We decided to finish this and use it for now. We should be using more native solutions on ARM macs, which could be solved by switching to nix native environment. Let's discuss this next year @tsusanka.

**Comment by tsusanka** [comment]
trezor/trezor-firmware#2025 merged!

**Comment by vdovhanych** [comment]
> [trezor/trezor-firmware#2025](https://github.com/trezor/trezor-firmware/pull/2025) merged!

@tsusanka This should be good to go then.

File: ci/build.yml [hunk 1]
```diff
@@ -34,6 +34,38 @@ build and push test image:
...
+  stage: build
+  <<: *build
+  when: manual
+  except:
+    - schedules
+    - master
+    - /^run\//
+  script:
+    - docker build  --tag $CONTAINER_NAME_ARM:CI_COMMIT_BRANCH-test -f docker/arm/Dockerfile .
+    - docker push $CONTAINER_NAME_ARM:CI_COMMIT_BRANCH-test
```
Comment (tsusanka) [inline ci/build.yml:67]: Let's remove this, we are reworking this in https://github.com/trezor/trezor-user-env/pull/127 anyway.

Comment (tsusanka) [reply ci/build.yml:67]: You meant `<<: *build_arm` anyway :).

Let's remove the anchor altogether then.

File: docker/arm/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,39 @@
+FROM vdovhanych/nixos
+
+# "fake" dbus address to prevent errors
+# https://github.com/SeleniumHQ/docker-selenium/issues/87
+#ENV DBUS_SESSION_BUS_ADDRESS=/dev/null
```
Comment (tsusanka) [inline docker/arm/Dockerfile:5]: Let's remove this.

File: src/binaries/trezord-go/bin/download.sh [hunk 3]
```diff
@@ -5,3 +5,4 @@ set -e
 
 # Bridge needs an older glibc so we are pinning to nixos-21.05 (stable) as of 2021-07-02
 nix-shell -p autoPatchelfHook -I "nixpkgs=https://github.com/NixOS/nixpkgs/archive/e9148dc1c30e02aae80cc52f68ceb37b772066f3.tar.gz" --run "autoPatchelf trezord-*"
+nix-shell -p autoPatchelfHook  --run "patchelf --set-interpreter /nix/store/0y64vbl8hc0xqq5ag6wwj5r3pr0l6z2s-glibc-2.33-56/lib/ld-linux-aarch64.so.1 ./src/binaries/trezord-go/bin/*-arm64"
```
Comment (tsusanka) [inline src/binaries/trezord-go/bin/download.sh:8]: Hardcoding the path is definitely wrong. We can't autopatch it? @mmilata might help here.

**Review by tsusanka — COMMENTED** [review]
I have added some comments. This will need more love in future but as a PoC it is great :).

File: ci/build.yml (hunk 1 above)
Comment (vdovhanych) [reply ci/build.yml]: removed it in c0220b4

File: docker/arm/Dockerfile (hunk 2 above)
Comment (vdovhanych) [reply docker/arm/Dockerfile]: removed in c0220b4

File: src/binaries/trezord-go/bin/download.sh (hunk 3 above)
Comment (vdovhanych) [reply src/binaries/trezord-go/bin/download.sh:8]: I think if we change it to `nix-shell -p autoPatchelfHook  --run "patchelf --set-interpreter /lib/ld-linux-aarch64.so.1 ./src/binaries/trezord-go/bin/*-arm64"` it should work and use the current store path

Comment (vdovhanych) [reply src/binaries/trezord-go/bin/download.sh:8]: Ok so we can't, it's not working without setting it like that.

**Comment by vdovhanych** [comment]
Closing this as we decided that multiarch build of docker image would be better.

**Pull request closed without merging**